pull_request_template.md

```diff
@@ -0,0 +1,22 @@
+<!--
+Resources:
+* contributing guidelines: https://catalystcoop-pudl.readthedocs.io/en/latest/CONTRIBUTING.html
+* code of conduct: https://catalystcoop-pudl.readthedocs.io/en/latest/code_of_conduct.html
+-->
+# Overview
```

Comment:
I would maybe add a section called # Documentation that encourages people to explain what they did and add to the `release notes` doc.

Reply:
Not every project has release notes, and this PR template only applies for projects that don't already have PR templates - so I just added a checklist item of "update any relevant documentation" to encourage that. Otherwise, I think the "explain what they did" stuff is what I meant by "What did you change?" Clarified that a bit too.